### PR TITLE
fix: default column config should be DBConfig and not RuntimeConfig

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

**Comment by JohannesMessner** [comment]
Can we make this a non-breaking change for a few versions? And raise a warning?

**Comment by JoanFM** [comment]
> Can we make this a non-breaking change for a few versions? And raise a warning?

What is breaking in your opinion? I do not see what can be breaking?

File: docarray/index/backends/hnswlib.py [hunk 1]
```diff
@@ -188,6 +183,12 @@ class RuntimeConfig(BaseDocIndex.RuntimeConfig):
             }
         )
 
+    @dataclass
+    class RuntimeConfig(BaseDocIndex.RuntimeConfig):
+        """Dataclass that contains all "dynamic" configurations of HnswDocumentIndex."""
+
+        pass
```
Comment (JohannesMessner) [inline docarray/index/backends/hnswlib.py:190]: I believe there are things that we could put here, but that is for a different PR I suppose

File: docs/user_guide/storing/index_hnswlib.md [hunk 2]
```diff
@@ -58,36 +58,31 @@ To load existing data, you can specify a directory that stores data from a previ
...
-            'ef': 15,
-            'M': 8,
-            'allow_replace_deleted': True,
-            'num_threads': 5,
-        },
-        None: {},
-    }
-)
+
+db = HnswDocumentIndex[MyDoc](work_dir='/tmp/my_db', default_column_config={
```
Comment (JohannesMessner) [inline docs/user_guide/storing/index_hnswlib.md:72]: can we `black` this?

**Review by JohannesMessner — APPROVED** [review]

**Pull request merged**